engine/src/main/java/org/terasology/world/block/family/BlockPlacementData.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.world.block.family;
+
+import org.joml.Vector2f;
+import org.joml.Vector2fc;
+import org.joml.Vector3f;
+import org.joml.Vector3fc;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+import org.terasology.math.Side;
+
+/**
+ * BlockPlacementData represents data that is useful for determining the orientation of new block.
+ * The data is supposed to be derived from the players state when placing the block.
+ * BlockPlacementData is immutable.
+ */
+public class BlockPlacementData {
+
+    public final Vector3ic blockPosition;
```

Comment:
will be done.

> Also, this will break some modules that define block families, right?

Since I kept the old `getBlockForPlacement()` method, nothing should break.

Reply:
here is the list of modules that need to be fixed:
```
Sample
MarcinScIncubator
MultiBlock
TutorialAssetSystem
Machines
Minesweeper
```
If you confirm the current API change, I'll roll out the fixes tomorrow.

Reply:
MarcinScIncubator still is a thing? 🤔 Thought we archived it?

@kBlaszczyk could you provide PRs for updating those modules (if not already happened), please 🙃

Reply:
"Thanks" to the way GitHub leaves redirects in place to moved repos it'll probably be easily possible to not realize something got archived ... :-) Just still works. Or in this case breaks.

Reply:
@skaldarnar yes, will do.